rocksdb_admin/admin_handler.cpp

```diff
@@ -281,6 +285,18 @@ bool SetAddressOrException(const std::string& ip,
   }
 }
 
+template <typename T>
+bool DecodeThriftStruct(const string& data, T* obj) {
+  auto ex = folly::try_and_catch<std::exception>([data, obj]() {
+    apache::thrift::CompactSerializer::deserialize(
```

Comment:
We may use BinarySerializer, I remember there were some compatible issues between Java and CPP for Compact protocol.

We may call deserialize() directly inside a try() catch {}.

Reply:
Done.
For my knowledge, why is try-catch better than the folly function?